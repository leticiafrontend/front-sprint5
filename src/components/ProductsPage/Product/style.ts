import styled from 'styled-components'

export const ProductCard = styled.li`
  @media (max-width: 1200px) {
    flex-basis: 50%;
  }
`

export const Card = styled.div`
  cursor: pointer;
  border: 1px solid white;
  padding: 8px;
  max-width: 250px;
  &:hover {
    box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2),
      0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0, 0, 0, 0.12);
  }
`

export const Img = styled.img`
  max-width: 100%;
  max-height: 100%;
`

export const Description = styled.p`
  font-size: 0.6875rem;
  font-weight: 400;
  color: rgba(0, 0, 0, 0.87);
  text-transform: uppercase;
  text-align: center;
  margin-top: 8px;
  margin-bottom: 10px;
`

export const Price = styled.p`
  font-size: 0.8125rem;
  font-weight: 600;
  text-transform: uppercase;
  text-align: center;
`
