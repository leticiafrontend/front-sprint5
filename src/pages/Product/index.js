import { SnackbarProvider } from 'notistack'
import React from 'react'

import { Header } from '../../components/Header/Header'
import { Loading } from '../../components/Loading/Loading'
import { ProductPage } from '../../components/ProductPage/ProductPage'

import { CategoriesContextProvider } from '../../context/CategoriesContext'
import { FilterContextProvider } from '../../context/FilterContext'
import { LoadingContextProvider } from '../../context/LoadingContext'
import { ProductsContextProvider } from '../../context/ProductsContext'

export const Product = () => (
  <>
    <SnackbarProvider>
      <LoadingContextProvider>
        <CategoriesContextProvider>
          <FilterContextProvider>
            <ProductsContextProvider>
              <Header />
              <ProductPage />
              <Loading />
            </ProductsContextProvider>
          </FilterContextProvider>
        </CategoriesContextProvider>
      </LoadingContextProvider>
    </SnackbarProvider>
  </>
)
