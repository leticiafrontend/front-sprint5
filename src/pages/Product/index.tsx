import React from 'react';
import { SnackbarProvider } from 'notistack';

import { Header } from '../../components/Header/Header';
import { Loading } from '../../components/Loading/Loading';
import { ProductPage } from '../../components/ProductPage/ProductPage';

import { CategoriesContextProvider } from '../../context/CategoriesContext';
import { FilterContextProvider } from '../../context/FilterContext';
import { LoadingContextProvider } from '../../context/LoadingContext';
import { ProductsContextProvider } from '../../context/ProductsContext';

export const Product: React.FC = () => (
  <>
    <SnackbarProvider>
      <LoadingContextProvider>
        <CategoriesContextProvider>
          <FilterContextProvider>
            <ProductsContextProvider>
              <Header />
              <ProductPage />
              <Loading />
            </ProductsContextProvider>
          </FilterContextProvider>
        </CategoriesContextProvider>
      </LoadingContextProvider>
    </SnackbarProvider>
  </>
);
