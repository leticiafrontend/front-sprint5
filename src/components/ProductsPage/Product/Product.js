import React from 'react'
import { Card, Description, Img, Price, ProductCard } from './style'

export const Product = ({ image, description, price }) => (
  <ProductCard>
    <Card>
      <Img src={image} />
      <Description>{description}</Description>
      <Price>{price}</Price>
    </Card>
  </ProductCard>
)
