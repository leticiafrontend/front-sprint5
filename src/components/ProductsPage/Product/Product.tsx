import React from 'react';
import { Card, Description, Img, Price, ProductCard } from './style';

interface PropsProduct {
  image: string;
  description: string;
  price: string;
}

export const Product: React.FC<PropsProduct> = ({
  image,
  description,
  price,
}) => (
  <ProductCard>
    <Card>
      <Img src={image} />
      <Description>{description}</Description>
      <Price>{price}</Price>
    </Card>
  </ProductCard>
);
