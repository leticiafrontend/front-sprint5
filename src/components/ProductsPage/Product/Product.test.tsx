import React from 'react'
import { render } from '@testing-library/react'
import { Product } from './Product'

describe('Componente de card de produtos', () => {
  it('O snapshot do componente deve permanecer sempre o mesmo', () => {
    const { container } = render(
      <Product
        image='/assets/images/c1.web'
        description='Jaqueta Puffer Juvenil Com Capuz Super Mario Branco Tam 10 a 18'
        price='199,90'
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
