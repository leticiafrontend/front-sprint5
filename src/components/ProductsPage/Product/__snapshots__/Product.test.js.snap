// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de card de produtos O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<li
  class="sc-bdnxRM gOHreT"
>
  <div
    class="sc-gtsrHT mNWRo"
  >
    <img
      class="sc-dlnjwi fYWFt"
      src="/assets/images/c1.web"
    />
    <p
      class="sc-hKFxyN hdqroc"
    >
      Jaqueta Puffer Juvenil Com Capuz Super Mario Branco Tam 10 a 18
    </p>
    <p
      class="sc-eCApnc rvNhV"
    >
      199,90
    </p>
  </div>
</li>
`;
